scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -361,7 +363,7 @@ public final DTLSFlight processMessage(Record record) throws HandshakeException
 			try {
 				record.setSession(session);
 				DTLSMessage messageToProcess = inboundMessageBuffer.getNextMessage(record);
-				while (nextFlight == null && messageToProcess != null) {
+				while (messageToProcess != null) {
```

Comment:
I am not sure if I understand correctly. You seem to suspect that there is the possibility of looping endlessly. Can you give an example of circumstances under which this might happen? I thought thought that we will always be able to break out of the loop simply because of the call to `doProcessMessage(Record)` which always *consumes* the message, thus removing it from the queue.